daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala

```diff
@@ -1621,6 +1621,49 @@ class EngineTest extends WordSpec with Matchers with EitherValues with BazelRunf
       err.msg should include("precondition violation")
     }
   }
+
+  "Engine#submit" should {
+    val cidV6 = toContractId("#cidV6")
+    val cidV7 = toContractId("#cidV7")
+    val contract = ValueRecord(
+      Some(Identifier(basicTestsPkgId, "BasicTests:Simple")),
+      ImmArray((Some[Name]("p"), ValueParty(party)))
+    )
+    val hello = Identifier(basicTestsPkgId, "BasicTests:Hello")
+    val templateId = TypeConName(basicTestsPkgId, "BasicTests:Simple")
+    val now = Time.Timestamp.now()
+    val submissionSeed = crypto.Hash.hashPrivateKey("engine check the version of input value")
+    def contracts = Map(
+      cidV6 -> ContractInst(templateId, VersionedValue(ValueVersion("6"), contract), ""),
+      cidV7 -> ContractInst(templateId, VersionedValue(ValueVersion("7"), contract), ""),
+    )
+
+    def run(cid: ContractId) = {
+      val engine = new Engine(EngineConfig.Stable)
+      val cmds = Commands(
+        submitter = party,
+        commands = ImmArray(
+          ExerciseCommand(templateId, cid, "Hello", ValueRecord(Some(hello), ImmArray.empty))),
+        ledgerEffectiveTime = now,
+        commandsReference = "",
+      )
+      engine
+        .submit(cmds, participant, submissionSeed)
+        .consume(contracts.get, lookupPackage, lookupKey)
+    }
+
+    "should succeeds if fed with allowed value version" in {
+      run(cidV6) shouldBe 'right
+    }
+
+    "should succeeds if fed with disallowed value version" in {
```

Comment:
Shouldn't we fail when we get something disallowed?
```suggestion
    "fail if fed with disallowed value version" in {
```

Reply:
good catch.